sdk/python/feast/infra/registry/proto_registry_utils.py

```diff
@@ -23,6 +24,25 @@
 from feast.stream_feature_view import StreamFeatureView
 
 
+def registry_proto_cache(func):
+    cache_key = None
+    cache_value = None
+
+    @wraps(func)
+    def wrapper(registry_proto: RegistryProto, project: str):
+        nonlocal cache_key, cache_value
+
+        key = tuple([id(registry_proto), registry_proto.version_id, project])
+
+        if key == cache_key:
+            return cache_value
+        else:
+            cache_value = func(registry_proto, project)
```

Comment:
Don't you need to update `cache_key` here as well?

Reply:
@achals I'm sorry, you're right. I added commit https://github.com/feast-dev/feast/pull/3702/commits/b8a373fa6cd48e0347322ed3685346c7ddb35f56